src/ripple/protocol/impl/STParsedJSON.cpp

```diff
@@ -542,18 +542,20 @@ static boost::optional<detail::STVar> parseLeaf (
 
                         std::string const strValue (account.asString ());
 
-                        if (value.size () == 40) // 160-bit hex account value
-                            uAccount.SetHex (strValue);
+                        // If we have what looks like a 160-bit hex account
+                        // value, try it and see if it's valid.
+                        bool const success = (value.size () == 40)
+                            ? uAccount.SetHex (strValue)
+                            : false;
 
+                        if (!success)
                         {
                             RippleAddress a;
-
-                            if (! a.setAccountID (strValue))
+                            if (!a.setAccountID (strValue))
```

Comment:
Habit - I don't like the space after the `!` (I know it's RippleD style) and must have done it unconsciously.